### DISCUSSION TITLE
Managing multiple PowerDNS servers from same Poweradmin instance?

### DISCUSSION THREAD

**Discussion by hevisko** [poweradmin/poweradmin#498] [Ideas]
I'm getting to a point where I have my main (internal) instance I use Poweradmin for, but now I have a need for an intermediary (for filtering/modifying AXFR transfers to go to public DNS) with some other domains. 

So the question is if it might be usefull for others to have poweradmin pointing to multiple powerdns servers, and be able to switch in the console without a separate poweradmin instance(s) for each powerdns server/instance?

**Comment by edmondas** [comment]
Recently, I have also been thinking about this topic. First I need to separate databases, since PowerDNS and PowerAdmin use the same database. So this could help us to have multiple connections to different PowerDNS databases, which makes DNSSEC implementation more difficult, but I think I can incrementally start using PowerDNS API. Ultimately, it will probably be a hybrid solution, but at some point it will be only an API-based solution. As a result, there will be no direct access to PowerDNS database, but some features may be removed (for example, list of domain letters).

**Comment by mkriegs** [comment]
Hey Edmondas,
can we go for more API Based UI ? 
best regards

**Comment by edmondas** [comment]
Currently, some progress is being made in this direction. As of now, I have tried to replace pdnsutil use with API calls, however, this is only working for one server. In addition, tenant support has not been implemented from the beginning, so supporting multiple servers might be more complex. As far as I can tell, this will be an incremental move, and a lot depends on how much development time can be allocated.

**Comment by edmondas** [comment]
Created a pre-release which allows DNSSEC to be handled through API:
https://github.com/poweradmin/poweradmin/releases/tag/v3.7.0-alpha.1

**Comment by edmondas** [comment]
Closing this since it has a tracking issue now. Multi-server management with a server switcher is #660. What did ship since this thread is API backend mode (4.3.0), where Poweradmin talks to PowerDNS over its API instead of the database, but it still manages one server per installation.
